packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +323,82 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("full initialization of data object should not happen by default", async () => {
+		const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[],
+		);
+
+		// Create a container for the first client.
+		const container1 = await provider.createContainer(runtimeFactory);
+		assert.doesNotThrow(
```

Comment:
This needs to be awaited otherwise, it may throw much later than this.
Change this to `await assert.doesNotReject`. Same with the other instances where you are doing this.